### PR TITLE
[Front] consume protected data

### PR DESCRIPTION
**Pull request by cedric25**

### CODE REVIEW

**Review by Le-Caignec — APPROVED** [review]
🚀

File: packages/sdk/src/lib/IExecDataProtectorModule.ts [hunk 1]
```diff
@@ -31,8 +31,16 @@ abstract class IExecDataProtectorModule {
...
     options?: DataProtectorConfigOptions
   ) {
+    const ipfsGateway = options?.ipfsGateway || DEFAULT_IPFS_GATEWAY;
+
     try {
-      this.iexec = new IExec({ ethProvider }, options?.iexecOptions);
+      this.iexec = new IExec(
+        { ethProvider },
+        {
+          ipfsGatewayURL: ipfsGateway,
```
Comment (cedric25) [inline packages/sdk/src/lib/IExecDataProtectorModule.ts:40]: Avoid duplicate params, use `ipfsGateway` given for DataProtector constructor.

File: packages/sdk/src/lib/dataProtectorCore/protectData.ts [hunk 2]
```diff
@@ -74,6 +75,8 @@ export const protectData = async ({
       isDone: true,
     });
 
+    vData = await serialiseDataIfNeeded(vData);
```
Comment (cedric25) [inline packages/sdk/src/lib/dataProtectorCore/protectData.ts:78]: Serialize file to Uint8Array.

File: packages/sdk/src/lib/types/coreTypes.ts [hunk 3]
```diff
@@ -14,7 +14,8 @@ export type DataScalarType =
   | bigint
   | string
   | Uint8Array
-  | ArrayBuffer;
+  | ArrayBuffer
+  | File;
```
Comment (cedric25) [inline packages/sdk/src/lib/types/coreTypes.ts:18]: Accept File, don't ask builder to encode it themselves into ArrayBuffer or Uint8Array.

**Pull request merged**